Forward port of 2.20x Performance Enhancements

andrewkfiedler opened the pull request.

andrewkfiedler committed:
[2.20.x] DDF-5904 Adds readonly translation for USNG / DMS to Lines / Polygons (#5905)

* DDF-5904 Adds readonly translation for USNG / DMS to Lines / Polygons

 - Adds a read only popup translation of USNG / MGRS to Lines and Polygons.  Similar with DMS.
 - In the future, we'll support the ability to enter coordinates in this format as well, but for now users at least need the ability to see what the translation from decimal degrees to these formats is.

* DDF-5904 Extract drawing model to it's own singleton

 - It used to be wedged into the store model, but that caused components that depended on drawing to inadvertantly have network dependencies. This pulls it into a self contained component that doesn't have as many dependencies (just backbone, wreqr and jquery).

a-asaad committed:
DDF-5909 Clear geo search drawing correctly when using query-basic form (#5910)

stustison committed:
DDF-5900 moved SubjectUtils, Permissions, and SecurityLogger to services

Commit:
Updated version ranges of pac4j and nimbus to work with new ddf versions

Commit:
Removed solr client reference (inlined constants) and added commons-math dependency

The commons-math3 dependency was previously being pulled in via solr/lucene transitives.
it is now explicit.

andrewkfiedler committed:
Updates how sources / statuses are handled for searches

 - This is the backend portion of the 2.20.x forward port.

andrewkfiedler committed:
Updates most of the frontend to the new result format

 - Left out only the table components for now.

andrewkfiedler committed:
save point

andrewkfiedler committed:
save search as visual

andrewkfiedler committed:
Save off this point

andrewkfiedler committed:
Save off

andrewkfiedler committed:
Readd grouping ability and update styling to match mocks

andrewkfiedler committed:
delete leftover code

andrewkfiedler committed:
Refactor destruction to be simpler, and handle nested groups gracefully

andrewkfiedler committed:
Uncomment cql write / read logic around nots

andrewkfiedler committed:
Remove a lot of references to workspaces / search forms / result forms

andrewkfiedler committed:
Roll up and simplify shell logic into app

andrewkfiedler committed:
remove redundant switch and add clarifying comment on router setup

andrewkfiedler committed:
Readd settings and fix issues with it

andrewkfiedler committed:
Convert nav buttons to real links

andrewkfiedler committed:
Update nav / settings to use real links

andrewkfiedler committed:
Add some nice touches to the theme for users

 - Theme settings moves out of the way when editing so users can see what they're changing, and backdrop opacity is removed

andrewkfiedler committed:
Deleting deleting deleting

andrewkfiedler committed:
Add not buttons and their stylings, update date components

andrewkfiedler committed:
add examples of current filter structure for reference

andrewkfiedler committed:
Convert core to use filter tree directly

andrewkfiedler committed:
Add animations

andrewkfiedler committed:
remove animations

andrewkfiedler committed:
Adjust the layout to make group ux better

andrewkfiedler committed:
Converts filter tree to map directly to components, all react based now

andrewkfiedler committed:
Adds new source selector, redesigns sort, changes how post sort / filtering works

andrewkfiedler committed:
redesign of settings for searches, adds back in navigation right but in lower left

andrewkfiedler committed:
Adds branding, ext to shell to allow new routes, changing of components, help url

 - also removes a restrictions on having more than one of each type of visual
 - fixes all instances of catalog-ui-search

andrewkfiedler committed:
Merge remote-tracking branch 'origin' into 2.20.x-ForwardPort

andrewkfiedler commented:
This probably won't work at the moment, there was a recent change to CqlRequestImpl with a merge conflict that I couldn't figure out at the moment.  If anyone wants to take a crack at it, the premerge change is here:
https://github.com/andrewkfiedler/ddf-ui/blob/2.20.x-ForwardPort/ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/CqlRequestImpl.java#L234
 and the new file is:
https://github.com/codice/ddf-ui/blob/master/ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/CqlRequestImpl.java

So in this PR, the lines from 178 to 200 in CqlRequestImpl need to be updated (removed, or moved?) before this will actually work.

andrewkfiedler committed:
Allow routes to not show in left nav

jlcsmith commented:
It appears that that logic in CqlRequestImpl was moved to https://github.com/codice/ddf-ui/blob/master/ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/QueryRequestBuilder.java#L177-L191

Commit:
resolves CqlRequestImpl merge issues

willwill96 commented on ui-frontend/lerna-debug.log:
```diff
@@ -0,0 +1,10 @@
+0 silly argv { _: [ 'run' ],
```
Should this be moved to the gitignore?

andrewkfiedler replied:
Yeah, not sure how it snuck in 😂

andrewkfiedler committed:
Bump mvn version of yarn

andrewkfiedler committed:
Update yarn locks

andrewkfiedler committed:
Force install

andrewkfiedler committed:
Adds postinstall step to root package

 - Before, we used workspaces which did this for us, but now we need to call lerna and bootstrap the packages ourselves.

andrewkfiedler committed:
Update circle ci timeout

andrewkfiedler committed:
Merge remote-tracking branch 'origin' into 2.20.x-ForwardPort
